core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2073,6 +2073,37 @@ package object config {
       .timeConf(TimeUnit.MILLISECONDS)
       .createOptional
 
+  private[spark] val SPECULATION_EFFICIENCY_ENABLE =
+    ConfigBuilder("spark.speculation.efficiency.enabled")
+      .doc("When set to true, spark will evaluate the efficiency of task processing through the " +
+        "stage task metrics and only need to speculate the inefficient tasks. A task is " +
+        "inefficient when its data process rate is less than the average data process " +
+        "rate of all successful tasks in the stage multiplied by a multiplier.")
+      .version("3.4.0")
+      .booleanConf
+      .createWithDefault(true)
+
+  private[spark] val SPECULATION_EFFICIENCY_TASK_PROCESS_MULTIPLIER =
+    ConfigBuilder("spark.speculation.efficiency.processMultiplier")
+      .doc("A multiplier for evaluating the efficiency of task processing. A task is inefficient " +
+        "when its data process rate is less than the average data process rate of all " +
+        "successful tasks in the stage multiplied by the multiplier.")
```

Comment:
"A multiplier that used when evaluating inefficient tasks. The higher the multiplier is, the more tasks will be possibly considered as inefficient."